examples/tp/tm_high_order.py

```diff
@@ -0,0 +1,527 @@
+    predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]
+    actColState = ['1' if i in activeColumnsIndeces else '0' for i in range(tm.numberOfColumns())]
+    actColStr = ("".join(actColState))
+    currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+    acc = accuracy(currCols, predCols)
+    x.append(ts)
+    y.append(acc)
+    ts += 1
+    predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+    predColState = ['1' if i in predictedColumnIndeces else '0' for i in range(tm.numberOfColumns())]
+    predColStr = ("".join(predColState))
+
+plt.ylim([-0.1,1.1])
+plt.plot(x, y)
+plt.xlabel("Timestep")
+plt.ylabel("Prediction Accuracy")
+plt.title("Fig. 1: TM learns sequence ABCD")
+plt.savefig("figure_1")
+plt.close()
+
+print ""
+print "-"*50
+print "Once the TM has learned the sequence ABCD, we will present the individual"
+print "characters to the TM to know its prediction. The TM outputs the columns"
+print "that become active upon the presentation of a particular character as well"
+print "as the columns predicted in the next time step. Here, you should see that"
+print "A predicts B, B predicts C, C predicts D, and D does not output any"
+print "prediction."
+print "N.B. Here, we are presenting individual characters, that is, a character"
+print "deprived of context in a sequence."
+print "-"*50
+print ""
+
+for k in range(4):
+  print "--- " + "ABCD"[k] + " ---"
+  tm.compute(set(seqT[k][:].nonzero()[0].tolist()), learn=False)
+  activeColumnsIndeces = [tm.columnForCell(i) for i in tm.getActiveCells()]
+  predictedColumnIndeces = [tm.columnForCell(i) for i in tm.getPredictiveCells()]
+  currCols = [1 if i in activeColumnsIndeces else 0 for i in range(tm.numberOfColumns())]
+  predCols = [1 if i in predictedColumnIndeces else 0 for i in range(tm.numberOfColumns())]
+  print("Active cols: " + str(np.nonzero(currCols)[0]))
+  print("Predicted cols: " + str(np.nonzero(predCols)[0]))
+  print ""
+
+# PART 2. Now we will present the sequence XBCY to the TM. The prediction accuracy
+# drops as the TM learns the new sequence. ABCD and XBCY are high-order sequences,
+# if we presented the sequence BC to the TM, it would not be able to disambiguate
+# between ABCD and XBCY. As such, we would expect to see simultaneous predictions
+# when presenting the TM with particular individual characters.
+print ""
+print "-"*50
+print "Part 2. We now present the sequence XBCY to the TM. As expected, the accuracy will"
+print "drop until the TM learns the new sequence (Fig 2). What will be the prediction of"
+print "the TM if presented with the sequence BC? This would depend on what character"
+print "anteceding B. This is an important feature of high-order sequences."
+print "-"*50
+print ""
+
+for t in range(10):
+  tm.reset()
+  for k in range(4):
+    tm.compute(set(seq2[k][:].nonzero()[0].tolist()), learn=True)
```

Comment:
There is a lot of repeated code in the loops with various "magical" statements. It is also hard to understand (for a newcomer) exactly what is going on in the loops. Can you create one or two functions that contain the repeated code, and document those functions well?  These loops can then just call those functions.